model/example_domain_test.go

```diff
@@ -0,0 +1,175 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleDomain_add() {
+	d1 := model.Multiple(1, 3, 5)
+	d2 := d1.Add(2, 4)
+	fmt.Println(d1)
+	fmt.Println(d2)
+	// Output:
+	// {[{1 1} {3 3} {5 5}]}
+	// {[{1 5}]}
+}
+
+func ExampleDomain_atLeast() {
+	d1 := model.NewDomain(
+		model.NewRange(1, 10),
+		model.NewRange(101, 110),
+	)
+	d2 := d1.AtLeast(50)
+	fmt.Println(d1)
+	fmt.Println(d2)
+	// Output:
+	// {[{1 10} {101 110}]}
+	// {[{101 110}]}
+}
+
+func ExampleDomain_atMost() {
+	d1 := model.NewDomain(
+		model.NewRange(1, 10),
+		model.NewRange(101, 110),
+	)
+	d2 := d1.AtMost(50)
+	fmt.Println(d1)
+	fmt.Println(d2)
+	// Output:
+	// {[{1 10} {101 110}]}
+	// {[{1 10}]}
+}
+
+func ExampleDomain_cmp() {
+	d1 := model.NewDomain(
+		model.NewRange(1, 5),
+		model.NewRange(8, 10),
+	)
+	d2 := model.Multiple(-1, 1)
+	fmt.Println(d1.Cmp(d2))
+	// Output:
+	// 1
+}
+
+func ExampleDomain_contains() {
+	d := model.NewDomain(model.NewRange(1, 10))
+	fmt.Println(d.Contains(5))
+	fmt.Println(d.Contains(15))
+	// Output:
+	// true
+	// false
+}
+
+func ExampleDomain_empty() {
+	d1 := model.NewDomain()
+	d2 := model.Singleton(42)
+	fmt.Println(d1.Empty())
+	fmt.Println(d2.Empty())
+	// Output:
+	// true
+	// false
+}
```

Comment:
We are missing `func ExampleDomain_iterator` to demonstrate use of `Domain.Iterator`.

Reply:
Added